### PR TITLE
Remove implicit credentials for Boto to fall back on

### PR DESCRIPTION
**Pull request by viren-nadkarni**
## Background

LocalStack currently sets (overrides) the Boto credentials and region during startup. The original intention was to handle cross-account usecases but since then, cross-account/region has been reworked to use explicit credentials obtained via request context.

Technically, the fallback from the environment is no longer required. Furthermore, by removing this fallback, we force future contributions use request context credentials programmatically. Same for external components that may make use of these env vars (such as QLDB). 

## Changes

- Removes access key ID and secret access key being set at startup.
- Removes some dead code.
- Pass proper test credentials to `test_terraform.py`. This went undetected because `terraform` would use the env values set by LocalStack startup.
- Maintain backward compatibility with init hooks. Upcoming major release will make init hook scripts require explicit credentials

## Related

Closes https://github.com/localstack/localstack/issues/10233

### CODE REVIEW

File: localstack/services/infra.py [hunk 1]
```diff
@@ -308,10 +293,7 @@ def prepare_environment():
 
         # set environment
         os.environ["AWS_REGION"] = AWS_REGION_US_EAST_1
```
Comment (viren-nadkarni) [inline localstack/services/infra.py:295]: Remove this too

File: localstack/constants.py [hunk 2]
```diff
@@ -5,12 +5,6 @@
 # LocalStack version
 VERSION = localstack.__version__
 
-# constant to represent the "local" region, i.e., local machine
-REGION_LOCAL = "local"
-
-# dev environment
-ENV_DEV = "dev"
-
```
Comment (viren-nadkarni) [inline localstack/constants.py:13]: The Environment logic was removed in https://github.com/localstack/localstack/pull/9644

File: localstack/constants.py [hunk 3]
```diff
@@ -5,12 +5,6 @@
 # LocalStack version
 VERSION = localstack.__version__
 
-# constant to represent the "local" region, i.e., local machine
-REGION_LOCAL = "local"
-
```
Comment (viren-nadkarni) [inline localstack/constants.py:10]: 'REGION_LOCAL' logic was removed in https://github.com/localstack/localstack/pull/9750

**Review by thrau — COMMENTED** [review]
> Technically, the fallback from the environment is no longer required.

does this imply that there are some other situations where it may be required for UX purposes? i don't think i understand the implications well enough to give a meaningful review.

maybe @dfangl can also take a peek

**Comment by dfangl** [comment]
I agree that we could avoid setting it now. If the tests pass, it is (manually) specified everywhere. However, this affects child processes which inherit the parents (LocalStacks) environment. I think for this change we should execute -ext tests before merging them, as it could have an unexpected repercussions.

Another thing this might influence are init scripts - for shell scripts it inherits the environment, and for python code it is executed in-process anyway. E.g. using the aws cli will probably break by default. For those it could make sense to continue setting them - otherwise it is probably a breaking change.

**Comment by viren-nadkarni** [comment]
@dfangl ext is tested working, see localstack/localstack-ext#2639

**Comment by dfangl** [comment]
Ah yes, somehow missed it!

**Review by alexrashed — COMMENTED** [review]
I agree with @dfangl. I had a quick look and to the best of my knowledge, this change is looking good for as we can make sure that we don't break any init scripts.
@viren-nadkarni Can you check the implication of this change on the different types of [init hooks](https://docs.localstack.cloud/references/init-hooks/) and how we might be able to mitigate them?
The code is basically all here: https://github.com/localstack/localstack/blob/master/localstack/runtime/init.py

**Comment by viren-nadkarni** [comment]
Thanks for the pointers!

With 04c3c589faa54100c4791a5b382c38fd6ef927f7, the init scripts will have access to the config vars as now. I think it makes sense to make the credentials a requirement within the init scripts in the next major release. I've marked this as so.

**Review by alexrashed — APPROVED** [review]
Thanks for addressing the comments! The change looks good to me now, let's make sure to remove the vars for the init hooks with the next major release. 🧹

**Pull request merged**